process/block/preprocess/intermediateResults.go

```diff
@@ -50,16 +50,17 @@ func NewIntermediateResultsProcessor(
 		adrConv:          adrConv,
 	}
 
-	irp.mutInterResultsForBlock.Lock()
 	irp.interResultsForBlock = make(map[string]*txInfo, 0)
-	irp.mutInterResultsForBlock.Unlock()
 
-	return nil, nil
+	return irp, nil
 }
 
 // CreateAllScrMiniBlocks returns the cross shard miniblocks for the current round created from the smart contract results
 func (irp *intermediateResultsProcessor) CreateAllInterMiniBlocks() []*block.MiniBlock {
 	miniBlocks := make([]*block.MiniBlock, irp.shardCoordinator.NumberOfShards())
+	for i := uint32(0); i < irp.shardCoordinator.NumberOfShards(); i++ {
```

Comment:
Is it really needed to create miniblocks for all shards?

Reply:
easier for creation and verification